homeassistant/components/light/tplink.py

```diff
@@ -149,22 +147,30 @@ def update(self):
         from pyHS100 import SmartDeviceException
         try:
             self._available = True
+
             if self._supported_features == 0:
                 self.get_features()
+
             self._state = (
                 self.smartbulb.state == self.smartbulb.BULB_STATE_ON)
-            if self._name is None:
+
+            # Pull the name from the device if a name was not specified
+            if self._name == DEFAULT_NAME:
```

Comment:
What is the point of `DEFAULT_NAME` if it is always overwritten?

Reply:
If the device is offline when HA starts this avoids the generic `unnamed device` default name.

Reply:
Okay, I didn't think of that scenario since my switches are always on.

I guess you could consider using `raise PlatformNotReady` in that situation, to avoid having to pick a name at all.

Reply:
Just update the name every time. Home Assistant will pick it up and update it.

Reply:
@amelchio Personally I prefer to see my devices as unavailable rather than them just not show up (which often happens with smart bulbs that are just on the edge of WiFi range). However I also name my devices manually so have no strong preference either way on this.

@balloob Retrieving the name every update will generate additional API queries to each device.

Reply:
Since the hosts are hardcoded, I think it's fine to show them as not available (because we know they exist).

Reply:
Yeah I am not really objecting :)